cmd/agent/app/processors/thrift_processor.go

```diff
@@ -104,8 +108,10 @@ func (s *ThriftProcessor) Stop() {
 func (s *ThriftProcessor) processBuffer() {
 	for readBuf := range s.server.DataChan() {
 		protocol := s.protocolPool.Get().(thrift.TProtocol)
-		protocol.Transport().Write(readBuf.GetBytes())
+		payload := readBuf.GetBytes()
+		protocol.Transport().Write(payload)
 		s.server.DataRecd(readBuf) // acknowledge receipt and release the buffer
+		s.logger.Debug("Span(s) received by the agent", zap.Int("bytes-received", len(payload)))
```

Comment:
potentially a race condition, better to log before releasing the buffer in `s.server.DataRecd(readBuf)`